src/validators/form-validator.ts

```diff
@@ -3,15 +3,18 @@ import { isTrue } from '../utils/boolean-utils';
 
 export const fieldRequiredErrCode = 'field.required';
 export const fieldOutOfBoundErrCode = 'field.outOfBound';
+export const fieldConditionalRequiredErrCode = 'field.conditionalRequired';
 
 export const FieldValidator: FormFieldValidator = {
-  validate: (field: FormField, value: any) => {
+  validate: (field: FormField, value: any, formValues: Record<string, any> ) => {
     if (field['submission']?.unspecified) {
       return [];
     }
-    if (isTrue(field.required) || isTrue(field.unspecified)) {
-      if (isEmpty(value)) {
-        return addError(fieldRequiredErrCode, 'Field is mandatory');
+    if (isEmpty(value)) {
+      if (  (typeof field.required === 'string' || typeof field.required === 'boolean' && isTrue(field.required)) || isTrue(field.unspecified)) {
```

Comment:
In the current implementation we do something like:

```ts
  if (isTrue(field.required) || isTrue(field.unspecified)) {
      if (isEmpty(value)) {
        return addError(fieldRequiredErrCode, 'Field is mandatory');
      }
    }

```

If a field has an empty value, we check whether it's required or unspecified. @ebambo it's not clear what the role of "unspecified" is. Does it simply let you bypass the mandatory behaviour of a field? Is it a implicit way of marking a field as required? Or both of the above?

Reply:
@samuelmale the `unspecified` is a way of bypassing the mandatory behaviour of a filed. This is to provide the user the ability of not filling a particular field if no information is available on the paper form. For some fields you can give clear options when there is missing information and you need to know why the missing information. An alternative to this would be not marking a field as required but, in this case, you won't know if the user did not fill the filed because he just missed the field or there was no information available. With unspecified if a particular field is empty you know that was the user decision not to fill that field. Hope this is clear

Reply:
So in summary, this property just let's us bypass the mandatory behaviour but not mark a field as mandatory. Eg. The field below shouldn't be required. 

```json
{
  "label": "Test Results",
  "id": "__L42EMuowp",
  "questionOptions": {
    "rendering": "select",
    "concept": "192f67ac-247c-4e1d-966a-4c223e33d38b",
    "answers": [
    ]
  },
  "required": false,
  "unspecified": true,
  "type": "obs"
}

```

Hence we should be doing:

```ts

  if (isTrue(field.required) /**|| isTrue(field.unspecified) **/) {
      if (isEmpty(value)) {
        return addError(fieldRequiredErrCode, 'Field is mandatory');
      }
    }

```

cc: @reagan-meant